miio/airfresh_t2017.py

```diff
@@ -134,32 +162,60 @@ def control_speed(self) -> int:
     @property
     def dust_filter_life_remaining(self) -> int:
         """Remaining dust filter life in percent."""
-        return self.data["filter_intermediate"]
+        if (
+            "filter_intermediate" in self.data
+            and self.data["filter_intermediate"] is not None
+        ):
+            return self.data["filter_intermediate"]
+
+        # Filter property of the A1
+        if "filter_rate" in self.data and self.data["filter_rate"] is not None:
+            return self.data["filter_rate"]
 
     @property
     def dust_filter_life_remaining_days(self) -> int:
         """Remaining dust filter life in days."""
-        return self.data["filter_inter_day"]
+        if (
+            "filter_inter_day" in self.data
+            and self.data["filter_inter_day"] is not None
+        ):
+            return self.data["filter_inter_day"]
```

Comment:
See my previous comment on checking for both values at once. This applies also for other properties that I will not comment separately to keep the review shorter.